### PR TITLE
Add details about materialized views in Iceberg

### PR DESCRIPTION
**Pull request by mosabua**
@anjalinorwood I used some of your info from the comments on the m views doc PR. 

Also note that all the other subsections are now moved under SQL support.

Screen Shot 2021-07-20 at 15 00 23

### CODE REVIEW

**Review by anjalinorwood — APPROVED** [review]
LGTM!

**Comment by mosabua** [comment]
Rebased to get the views support PR that @findepi merged accounted for. Ready to go again. Would be great to get it merged @electrum

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 1]
```diff
@@ -91,13 +91,14 @@ supports the following features:
...
-* :ref:`sql-materialized-views-management`
+* :ref:`sql-materialized-views-management`, see also
+  :ref:`iceberg-materialized-views`
 * :ref:`sql-views-management`
 
 .. _iceberg-tables:
 
 Partitioned tables
-------------------
+^^^^^^^^^^^^^^^^^^
```
Comment (findepi) [inline docs/src/main/sphinx/connector/iceberg.rst:101]: unrelated changes

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 2]
```diff
@@ -226,3 +229,29 @@ and a file system location of ``/var/my_tables/test_table``::
...
+Materialized views
+^^^^^^^^^^^^^^^^^^
+
+The Iceberg connector supports :ref:`sql-materialized-views-management`. In the
+underlying system each materialized view consists of a view definition and an
+Iceberg storage table. The storage table name is stored as a materialized view
+property. The data is stored in that storage table.
+
+You can use the :ref:`iceberg-table-properties` to control the created storage
+table and therefore the layout and performance. For example, you can use the
```
Comment (findepi) [inline docs/src/main/sphinx/connector/iceberg.rst:244]: Do we have a list of table properties that are supported for a materialized view?

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 3]
```diff
@@ -226,3 +229,29 @@ and a file system location of ``/var/my_tables/test_table``::
...
+following clause with :doc:`/sql/create-materialized-view` to use the ORC format
+for the data files and partition the storage per day using the column
+``_date``::
+
+    WITH ( format = 'ORC', partitioning = ARRAY['_date'] )
```
Comment (findepi) [inline docs/src/main/sphinx/connector/iceberg.rst:249]: leading underscores are not typical. Let's use `a_date`, `some_date` or `event_date`

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 4]
```diff
@@ -226,3 +229,29 @@ and a file system location of ``/var/my_tables/test_table``::
...
+
+Updating the data in the materialized view with
+:doc:`/sql/refresh-materialized-view` deletes the data from the storage table,
+and inserts new data that is the result of executing the materialized view
+query.
```
Comment (findepi) [inline docs/src/main/sphinx/connector/iceberg.rst:254]: @sopel39 pease confirm.

Also, this could be understood as "during REFRESH, users may temporarily get zero rows results when querying the materialized view". We should be clear whether this may or may not be the case.

**Comment by findepi** [comment]
cc @alexjo2144

File: docs/src/main/sphinx/connector/iceberg.rst (hunk 4 above)
Comment (sopel39) [reply docs/src/main/sphinx/connector/iceberg.rst:254]: > @sopel39 pease confirm.

Yes, in `finishRefreshMaterializedView` we first drop data from storage table and then commit new data

Comment (findepi) [reply docs/src/main/sphinx/connector/iceberg.rst:254]: @mosabua please add a warning (or sth) about this in the docs.

File: docs/src/main/sphinx/connector/iceberg.rst (hunk 1 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/iceberg.rst:101]: It is not unrelated - it establishes the Materialized views section and all others under the SQL support

File: docs/src/main/sphinx/connector/iceberg.rst (hunk 2 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/iceberg.rst:244]: The section it links to has that..

File: docs/src/main/sphinx/connector/iceberg.rst (hunk 3 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/iceberg.rst:249]: done

File: docs/src/main/sphinx/connector/iceberg.rst (hunk 4 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/iceberg.rst:254]: I don't think a warning is warranted as @anjalinorwood approved the PR as is.. it seems like a very minor detail with potentially no user impact. I dont know how it works out in practical terms either. E.g. if refresh takes long to populate the table. Is this really a problem or is access to the materialized view blocked during the refresh somehow?

What is your experience with lots of users on this @anjalinorwood ?

File: docs/src/main/sphinx/connector/iceberg.rst (hunk 1 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/iceberg.rst:101]: See screenshot in description .. without this the order of sections and hierarchy makes no sense

**Review by findepi — CHANGES_REQUESTED** [review]
lgtm % warning about non-atomic refresh

**Review by electrum — APPROVED** [review]

**Review by electrum — APPROVED** [review]

File: docs/src/main/sphinx/connector/iceberg.rst (hunk 4 above)
Comment (electrum) [reply docs/src/main/sphinx/connector/iceberg.rst:254]: We discussed and agreed that a warning is appropriate. It looks like it would be easy to change the code to do this as a single atomic Iceberg transaction, so we can then remove the warning.
